### PR TITLE
Sync all conversations syncs all

### PR DESCRIPTION
**Pull request by nplasterer**
Update the sync all conversations to also sync all welcomes. Also adds this to the web and makes sure they are syncing sync groups.

Fixes: https://github.com/xmtp/libxmtp/issues/1316

### CODE REVIEW

File: bindings_ffi/src/mls.rs [hunk 1]
```diff
@@ -917,19 +917,30 @@ impl FfiConversations {
...
             groups.len()
         );
 
-        let num_groups_synced: usize = inner.sync_all_groups(groups).await?;
-        // Uniffi does not work with usize, so we need to convert to u32
+        // Call sync operations in parallel
+        let (welcome_result, group_result) = tokio::join!(
+            self.inner_client.sync_welcomes(&conn),
+            inner.sync_all_groups(groups)
+        );
```
Comment (richardhuaaa) [inline bindings_ffi/src/mls.rs:933]: If we do these in parallel, is there a chance that we don't sync the messages on a new group?

File: bindings_ffi/src/mls.rs [hunk 2]
```diff
@@ -917,19 +917,30 @@ impl FfiConversations {
 
     pub async fn sync_all_conversations(&self) -> Result<u32, GenericError> {
```
Comment (richardhuaaa) [inline bindings_ffi/src/mls.rs:918]: Are there any clients depending on the current behavior? Is it worth making a separate method called `sync_all`, or renaming this one? Don't have a strong opinion

**Review by cameronvoell — APPROVED** [review]
nice improvement 👍

File: bindings_ffi/src/mls.rs (hunk 1 above)
Comment (nplasterer) [reply bindings_ffi/src/mls.rs:933]: O good call out!

**Review by insipx — APPROVED** [review]

**Pull request merged**